### PR TITLE
feat: add new blog post - turn the senior around

### PR DESCRIPTION
**Pull request by shaimendel**
The relevant doc to review:  docs/md/docs/2020-11-02-turn-the-senior-around.md.
Better to watch in https://github.com/shaimendel/dev-blog/blob/blog/turn_the_senior_around/docs/md/docs/2020-11-02-turn-the-senior-around.md.

### CODE REVIEW

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Turn The Senior Around
+
+As engineering managers we take many considerations into account, two of the main ones are:
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: Considerations are derived from managerial responsibilities. So I'd rephrase this sentence as:
"As Engineering Managers, we carry many responsibilities. two of which are:"

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- Delivering on time
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: I agree on-time delivery is important. How do you feel about its quality?

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- Make our people grow
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: The use of **_Make_** sounds like growing their team members is the Engineering Manager's responsibility. IMHO a rephrase of that sentence to "Support your team members professional growth and development." shits that towards the team members

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+During my experience as an engineering manager and as an engineer all the managers I have met saw those subjects critical and invested a lot in improving them.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: How does this sentence sound?
`These two responsibilities were considered critical amongst all the engineering managers I have met with throughout my career as an Engineering Manager and as a Software Engineer.`

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Another consensus among them all was that senior engineers are a key enabler to both topics, they are the main players the team lead can rely on to achieve those goals.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: A couple of suggestions:
- Break this sentence into two shorter sentences for better readability
- Replace `topics` with `responsibilities` to reflect that it is also a responsibility of every team member

`Another consensus among them all was that senior engineers are a key enabler to both responsibilities. They are the main players the team lead can rely on to achieve those goals.`

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+I spoke a bit about the role of senior engineer in my [The Last Dance blog](/docs/2020-07-09-last-dance) and this blog relates to it - I want to discuss about __*how*__ should engineering managers utilize their senior engineers to achieve those goals.
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: short rephrase `I spoke a bit about the senior engineer's role in my`

Comment (nirfuchs) [reply docs/md/docs/2020-11-02-turn-the-senior-around.md]: I'd replace `utilize` as it may sound like senior engineers are a tool used by the engineering manager. how about the following: `I want to discuss how can engineering managers successfully achieve teams' goals by harnessing their senior engineers' power?`

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+Most of the engineering managers I have met used their senior engineers to lead all the critical features, I would like to turn the senior around and suggest the opposite - the more senior you are the the less you lead engineering efforts in the team.
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: I suggest replacing`used` with `assigned`, as used sounds more like senior engineers are a tool

Comment (nirfuchs) [reply docs/md/docs/2020-11-02-turn-the-senior-around.md]: `turn the senior engineer around` seems more precise to me

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+The more senior you are the more you promote other engineers by giving constant feedback on features __*they lead*__.
+
+Needless to say this only reflects my opinion :)
+
+# The Common Team Structure - Senior First Delivery
+![The common team structure](/docs/assets/turn-the-senior-around/common_way.png)<br><br>
+The common team structure I saw was senior engineers leading features, while trying to improve and grow the non senior members during the process.
+
+## How Does This Work
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md:20]: This is a question. The question mark should come at the end, please :)

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+The senior engineers are the ones responsible to make the feature delivered, they are the ones with all the context and they are the active leaders of those feature squads - communication with stakeholders, design etc.<br>
+The team lead's job is to make sure the senior engineers are on track and help them if needed.<br>
+The advantages are obvious: this way there is maximal certainty that the features will be delivered on time, the non senior members learn by watching the senior in action and by (hopefully) active mentoring.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md:23]: I like that you chose `advantages` to describe things that work well by choosing this team structure. For consistency, I'd replace the use of `issues` later in the article with `disadvantages`.

Another way to go about it here would have been using the terms `pros` & `cons`

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+You see a team where senior engineers can break through the senior limits and get promoted themselves.<br>
+You see a team with satisfied team members :)<br>
+
+## The Risk
+Having said all of that there is a short term risk here - you put all your faith and trust on the non senior members to lead critical features and rely mostly on the seniors to help them.<br>
+This requires a huge trust in the team and an ability of the team lead to maximize the empowerment of the team.<br>
+I personally really enjoy this step, when you trust your team this way the team feels it - this ability to rely on the team while having long term team building vision is exactly where we grow as engineering managers, but this might be a topic to another blog post :) <br>
+
+# Summary
+Turning the senior around means to flip the team structure - let the non seniors lead features and make sure the senior members grow them by constant feedback.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md:67]: This sentence sounds like it could use some rephrasing.

> Turning the senior engineer around also means flipping team structure around by letting non-senior engineers lead features while senior engineers are helping them grow by constantly providing feedback.

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+This allows strong team building, maximize everyone's growth and eventually fulfill the two objectives at the beginning: delivering on time and making our team grow.
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: small corrections:
* `maximize` should be `maximizes`
* `fulfill` should be `fulfills`
> two objectives at the beginning

should become

> two objectives stated at the beginning

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+The common team structure I saw was senior engineers leading features, while trying to improve and grow the non senior members during the process.
+
+## How Does This Work
+The senior engineers are the ones responsible to make the feature delivered, they are the ones with all the context and they are the active leaders of those feature squads - communication with stakeholders, design etc.<br>
+The team lead's job is to make sure the senior engineers are on track and help them if needed.<br>
+The advantages are obvious: this way there is maximal certainty that the features will be delivered on time, the non senior members learn by watching the senior in action and by (hopefully) active mentoring.<br>
+
+## Issues With Senior First Delivery
+### Team Stagnation
+Working in this structure preserves the current hierarchy of the team. The seniors are the leaders, the non seniors are there to be lead. <br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: * `led` instead of `lead`

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+This by definition means that the squad members will always be as good as their senior squad leads - all they see is what the senior engineer shows them - in the happy flow by active mentoring, in the worst case just by watching them.
+
+### Limited Growth
+This is the key part in my opinion, both the seniors and the non seniors have limited amount of potential growth in this scenario, actually the growth is relatively really slow. <br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md:31]: I believe this sentence refers to `growth rate`, not just `growth`, right?

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+* From the senior engineer's perspective: they do more of the same - they became seniors because they know how to design, do decompositions, update stakeholders, code well.<br>
+But that not they're main purpose right? the senior engineers are measured by their impact on the group, by making others better. How can seniors be focused on improving others when they are focused on leading a feature?<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: A small rephrase:
> This structure misses the point where senior engineers are measured by their impact on the group and making others better. Instead, senior engineers are measured here by their ability to deliver the feature on-time.

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+So the senior engineers improve certain skills which they are already good at, but do not maximize the most important muscle - to make others better.
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: grow their most important muscle :)

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+* From the non senior engineer's perspective: they are expected to grow by watching and not to grow by doing. So yes - they will grow by watching the senior engineer leading the feature and by active mentoring by them, but they will improve less than doing that themselves and being exposed to all of the different aspects of leading a feature: having all the context, communication with stakeholders etc.
+
+# Turn The Senior Around
+![Turn The Senior Around](/docs/assets/turn-the-senior-around/new_way.png)<br><br>
+The non engineers are the ones responsible to make the feature delivered, they are the ones with all the context of the feature and they are the active leaders of those feature squads.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: 👍

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+The senior engineer's role is first and foremost to constantly give feedback to the non senior engineer who leads the feature - help them learn by providing them a great ground for them to grow.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md:40]: 💯

File: docs/md/docs/2020-11-02-turn-the-senior-around.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+The team lead's job is double:
+- Communicate with the non seniors as feature leads, help if needed and make sure everything is on track
+- Communicate with the senior engineers and give them all the help needed to __*make sure they have all they need to grow the non senior members*__. The team lead constantly gives feedback to the senior engineers on how they improve others.
+
+## Advantages Of Turning The Senior Around
+### Dynamic Team Hierarchy
+The non seniors feel that the limit to their performance is only themselves and not the senior who leads them (in the scenario before). The fact that the non seniors lead the feature provides them the needed experience and confidence to later become seniors themselves.
+
+### Maximal Growth
```
Comment (nirfuchs) [inline docs/md/docs/2020-11-02-turn-the-senior-around.md]: 💯

**Review by nirfuchs — APPROVED** [review]
This is a really nice article overall, I made some grammar and structure comments mostly.
would be a nice topic to discuss in our 1-1 😉

**Comment by shaimendel** [comment]
@nirfuchs thank you so much! :)
